powersimdata/input/grid.py

```diff
@@ -29,44 +29,20 @@ class Grid:
     :raises ValueError: if source or engine does not exist.
     """
 
-    def __init__(self, interconnect, source="usa_tamu", engine="REISE"):
+    def __init__(self, model_data):
         """Constructor."""
-        if not isinstance(source, str):
-            raise TypeError("source must be a str")
-        if source not in self.SUPPORTED_MODELS and not source.endswith(".mat"):
-            raise ValueError(
-                f"Source must be one of {','.join(self.SUPPORTED_MODELS)} "
-                "or the path to a .mat file that represents a grid "
-            )
-        if engine not in self.SUPPORTED_ENGINES:
-            raise ValueError(
-                f"Engine must be one of {','.join(self.SUPPORTED_ENGINES)}"
-            )
-
-        key = cache_key(interconnect, source)
-        cached = _cache.get(key)
-        if cached is not None:
-            data = cached
-        elif source == "usa_tamu":
-            data = TAMU(interconnect)
-        elif os.path.splitext(source)[1] == ".mat":
-            if engine == "REISE":
-                data = FromREISE(source)
-            elif engine == "REISE.jl":
-                data = FromREISEjl(source)
-
-        self.data_loc = data.data_loc
-        self.interconnect = data.interconnect
-        self.zone2id = data.zone2id
-        self.id2zone = data.id2zone
-        self.sub = data.sub
-        self.plant = data.plant
-        self.gencost = data.gencost
-        self.dcline = data.dcline
-        self.bus2sub = data.bus2sub
-        self.bus = data.bus
-        self.branch = data.branch
-        self.storage = data.storage
+        self.data_loc = model_data.data_loc
+        self.interconnect = model_data.interconnect
+        self.zone2id = model_data.zone2id
+        self.id2zone = model_data.id2zone
+        self.sub = model_data.sub
+        self.plant = model_data.plant
+        self.gencost = model_data.gencost
+        self.dcline = model_data.dcline
+        self.bus2sub = model_data.bus2sub
+        self.bus = model_data.bus
+        self.branch = model_data.branch
+        self.storage = model_data.storage
 
         _cache.put(key, self)
```

Comment:
I'm curious, why was this `_cache` system put in and what problem did it solve?

Reply:
It can take some time to instantiate a USA `Grid` class so we save the object in memory. To illustrate, in **PostREISE**, we do analysis/plotting based on the scenario id and we often create a `Grid` object in these functions. Instead of creating a `Grid` object for each analysis/plots of the same scenario, we load it from memory.

Reply:
One place it is useful is running tests - we end up using one of the base grids multiple times, but since it's the same each time (not caching transformed grids here), we can reuse a copy from memory. If I recall this led to a slight improvement in the time it takes to run the test suite.

Reply:
What are your opinions on pulling the cache out of the Grid class? This would:
- make the pulling out of the initialization of the `data` variable easier
- make the code simpler - the idea of a Cache class that manages Grid objects (or any other objects) makes more sense than a Grid class that holds an internal cache imo
Cons:
- I can see user experience going down. I imagine data scientists who work with Grid everyday in jupyter notebooks don't want to have to also remember to import a Cache class to put Grid into. I could see this potentially being alleviated by providing helper functions such as `cache.get_or_create(Grid, TAMU, ["Western"))` but it could potentially mess with people's workflows
- Finding an intuitive key for a Cache that is pulled out of the Grid class seems difficult to say the least and eludes me currently